sdk/src/main/scala/com/horizen/account/state/AccountStateView.scala

```diff
@@ -1,36 +1,164 @@
+            ForgerStakeSmartContractAddress,
+            BigInteger.ZERO, // gasPrice
+            BigInteger.ZERO, // gasFeeCap
+            BigInteger.ZERO, // gasTipCap
+            BigInteger.ZERO, // gasLimit
+            stakedAmount,
+            BigInteger.ONE.negate(), // a negative nonce value will rule out collision with real transactions
+            data)
+
+          val processor = messageProcessors.find(_.canProcess(message, this)).getOrElse{
+            val errMsg = s"No known processor for msg: $message"
+            log.error(errMsg)
+            throw new IllegalArgumentException(errMsg)
+          }
+
+          processor.asInstanceOf[ForgerStakeMsgProcessor].doAddNewStakeCmd(message, view, isGenesisScCreation = true) match {
+            case res: ExecutionFailed =>
+              log.error(res.getReason.getMessage)
+              throw new IllegalArgumentException(res.getReason)
+            case res: InvalidMessage =>
+              log.error(res.getReason.getMessage)
+              throw new IllegalArgumentException(res.getReason)
+            case _ : ExecutionSucceeded =>
+          }
+
+        case ft: ForwardTransfer =>
+          val ftOut : MainchainTxForwardTransferCrosschainOutput = ft.getFtOutput
+
+          // we trust the MC that this is a valid amount
+          val value = ZenWeiConverter.convertZenniesToWei(ftOut.amount)
+
+          // we must get 20 bytes out of 32 with the proper padding and byte order
+          // MC prepends a padding of 0 bytes (if needed) in the sc_create command when a 32 bytes address is specified.
+          // After reversing the bytes, the padding is trailed to the correct 20 bytes proposition
+          val recipientProposition = new AddressProposition(BytesUtils.reverseBytes(ftOut.propositionBytes.take(com.horizen.account.utils.Account.ADDRESS_SIZE)))
+
+          log.debug(s"adding FT amount = $value to address=$recipientProposition")
+
+          // TODO check we have this account, otherwise set codeHash??
+          view.addBalance(recipientProposition.address(), value)
+
+      })
+    })
+  }
+
+  def getOrderedForgingStakeInfoSeq() : Seq[ForgingStakeInfo] = {
+    val data: Array[Byte] = BytesUtils.fromHexString(AddNewStakeCmd)
+    val message = new Message(
+      null, // sender proposition
+      ForgerStakeSmartContractAddress,
+      BigInteger.ZERO, // gasPrice
+      BigInteger.ZERO, // gasFeeCap
+      BigInteger.ZERO, // gasTipCap
+      BigInteger.ZERO, // gasLimit
+      BigInteger.ZERO, // value, not used here,
+      BigInteger.ONE.negate(), // nonce, not used here
+      data)
+
+    val processor = messageProcessors.find(_.canProcess(message, this)).getOrElse{
+      val errMsg = s"No known processor for msg: $message"
+      log.error(errMsg)
+      throw new IllegalArgumentException(errMsg)
+    }
+
+    val forgerStakeList = processor.asInstanceOf[ForgerStakeMsgProcessor].doGetListOfForgersCmd(message, this) match {
```

Comment:
I have a perception that we can avoid constrcuting message and other.
Instead, we may add one more method to the `ForgerStakeMsgProcessor` that will ignore the msg check and gas payments. The original method may use it as well.
See, how we did for WithdrawalRequests

Reply:
This msg processor is not a singleton, we have to find it s instance in the message processor list, and the canProcess() has the msg as parameter. I added a tag in order to find it without building the msg obj, pls check if this approach is OK.

Reply:
Used an approach similar to WithdrawalRequests for getting the proper message processor